microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIConfigFileHealthTest.java

```diff
@@ -0,0 +1,204 @@
+                + "/metrics";
+        metricsRequest = given().baseUri(url).accept(ContentType.JSON);
+        // reset MP Config properties
+        new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();
+    }
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        return ShrinkWrap
+                .create(WebArchive.class, FailSafeCDIHealthBaseTest.class.getSimpleName() + ".war")
+                .addClasses(FailSafeDummyService.class, CDIBasedLivenessHealthCheck.class, CDIBasedReadinessHealthCheck.class)
+                .addAsManifestResource(new StringAsset(MPConfigContent), "microprofile-config.properties")
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    /**
+     * @tpTestDetails Multiple-component customer scenario - health check status is based on fail-safe CDI bean and MP Config
+     *                property. There is IOException since service is in maintenance but fallback method should be used.
+     * @tpPassCrit Overall and the health check status is down and fallback method is invoked. MP Metrics are increased
+     *             according to the specification.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Ignore("WFLY-12924, WFLY-12925")
+    @Test
+    final public void testHealthEndpointDownInMaintenace() throws Exception {
+        // TODO Java 11 Map<String, String> liveCheck = Map.of( "name", "dummyLiveness", "status", "DOWN");
+        Map<String, String> liveCheck = Collections.unmodifiableMap(new HashMap<String, String>() {
+            {
+                put("name", "dummyLiveness");
+                put("status", "DOWN");
+            }
+        });
+        Map<String, String> readyCheck = Collections.unmodifiableMap(new HashMap<String, String>() {
+            {
+                put("name", "dummyReadiness");
+                put("status", "DOWN");
+            }
+        });
+        get(HealthUrlProvider.healthEndpoint()).then()
+                .statusCode(503)
+                .contentType(ContentType.JSON)
+                .header("Content-Type", containsString("application/json"))
+                .body("status", is("DOWN"),
+                        "checks", hasSize(2),
+                        "checks", containsInAnyOrder(liveCheck, readyCheck));
+
+        MetricsChecker.get(metricsRequest)
+                .validateSimulationCounter(FailSafeDummyService.MAX_RETRIES + 1) // 1 call + N retries
+                .validateInvocationsTotal(1)
+                .validateInvocationsFailedTotal(1)
+                .validateRetryRetriesTotal(FailSafeDummyService.MAX_RETRIES) // N retries
+                .validateRetryCallsFailedTotal(1)
+                .validateRetryCallsSucceededTotal(0)
+                .validateFallbackCallsTotal(1);
+
+        get(HealthUrlProvider.healthEndpoint()).then().statusCode(503);
```

Comment:
I guess that headers and body validation is skipped for this one because you already did it on the previous call, right? In such case, what about a single line comment here?

Reply:
Exatly. I will add one.